internal/jobcontainers/jobcontainer.go

```diff
@@ -0,0 +1,445 @@
+	if matched, err := regexp.MatchString(volumeGUIDRegex, s.Root.Path); !matched || err != nil {
+		return nil, fmt.Errorf(`invalid container spec - Root.Path '%s' must be a volume GUID path in the format '\\?\Volume{GUID}\'`, s.Root.Path)
+	}
+	if s.Root.Path[len(s.Root.Path)-1] != '\\' {
+		s.Root.Path += `\` // Be nice to clients and make sure well-formed for back-compat
+	}
+
+	container := newJobContainer(id, s)
+
+	// Create the job object all processes will run in. We set true for the notifications argument
+	// as we always want to receive messages from the job.
+	job, err := jobobject.Create(ctx, id, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create job object")
+	}
+
+	// Parity with how we handle process isolated containers. We set the same flag which
+	// behaves the same way for a silo.
+	if err := job.SetTerminateOnLastHandleClose(); err != nil {
+		return nil, errors.Wrap(err, "failed to set terminate on last handle close on job container")
+	}
+	container.job = job
+
+	var path string
+	defer func() {
+		if err != nil {
+			container.Close()
+			if path != "" {
+				removeSandboxMountPoint(ctx, path)
+			}
+		}
+	}()
+
+	limits, err := specToLimits(ctx, s)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to convert OCI spec to job object limits")
+	}
+
+	// Set resource limits on the job object based off of oci spec.
+	if err := job.SetResourceLimits(limits); err != nil {
+		return nil, errors.Wrap(err, "failed to set resource limits")
+	}
+
+	// Setup directory sandbox volume will be mounted
+	sandboxPath := fmt.Sprintf(sandboxMountFormat, id)
+	if _, err := os.Stat(sandboxPath); os.IsNotExist(err) {
+		if err := os.MkdirAll(sandboxPath, 0777); err != nil {
+			return nil, errors.Wrap(err, "failed to create mounted folder")
+		}
+	}
+	path = sandboxPath
+
+	if err := mountSandboxVolume(ctx, path, s.Root.Path); err != nil {
+		return nil, errors.Wrap(err, "failed to bind payload directory on host")
+	}
+
+	container.sandboxMount = path
+	go container.waitBackground(ctx)
+	return container, nil
+}
+
+// CreateProcess creates a process on the host, starts it, adds it to the containers
+// job object and then waits for exit.
+func (c *JobContainer) CreateProcess(ctx context.Context, config interface{}) (_ cow.Process, err error) {
+	conf, ok := config.(*hcsschema.ProcessParameters)
+	if !ok {
+		return nil, errors.New("unsupported process config passed in")
+	}
+
+	path, args := separateArgs(conf.CommandLine)
+	absPath, err := findExecutable(path, c.sandboxMount)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find executable")
+	}
+
+	token, err := processToken(conf.User)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create process token")
+	}
+
+	env := append(os.Environ(), "CONTAINER_SANDBOX_MOUNT_POINT="+c.sandboxMount)
+	env = append(env, envMapToSlice(conf.Environment)...)
```

Comment:
Kinda small, but if the conf.Environment has a value for "CONTAINER_SANDBOX_MOUNT_POINT", wouldn't the one above be overwritten? Don't we want to do that the other way around?

Reply:
The hope is there won't be, but obviously that's not guaranteed, this would just have to be called out in docs. This is kind of a sore spot for this work as the user won't know where the mount point is before running the container as it's generated, but they need to know the path to it if they want to provide the path to, for instance, a powershell script in the image. This is what we had devised from working with folks who tried this internally.

Reply:
Reread this and see what you're saying. I'm not sure how Windows treats duplicate variables in the env block. Let me test this out.

Reply:
Looks like it takes whatever was set last, so reordered! Thanks for this!